src/app/clusters/electrical-energy-measurement-server/ElectricalEnergyMeasurementCluster.h

```diff
@@ -32,31 +33,104 @@ enum class OptionalAttributes : uint32_t
+                optionalAttributes.Has(ElectricalEnergyMeasurement::OptionalAttributes::kOptionalAttributeCumulativeEnergyReset) &&
+                featureFlags.Has(ElectricalEnergyMeasurement::Feature::kCumulativeEnergy));
+        }
+
+        EndpointId endpointId;
+        const BitFlags<ElectricalEnergyMeasurement::Feature> mFeatureFlags;
+        OptionalAttributesSet mEnabledOptionalAttributes;
+    };
+
+    ElectricalEnergyMeasurementCluster(const Config & config) :
+        DefaultServerCluster({ config.endpointId, ElectricalEnergyMeasurement::Id }), mFeatureFlags(config.mFeatureFlags),
+        mEnabledOptionalAttributes(config.mEnabledOptionalAttributes)
     {}
 
-    ~ElectricalEnergyMeasurementAttrAccess() { Shutdown(); }
+    const OptionalAttributesSet & OptionalAttributes() const { return mEnabledOptionalAttributes; }
+    const BitFlags<ElectricalEnergyMeasurement::Feature> & Features() const { return mFeatureFlags; }
+
+    // Direct access to measurement data - for backwards compatibility
+    const ElectricalEnergyMeasurement::MeasurementData * GetMeasurementData() const { return &mMeasurementData; }
+
+    // Getters - return copies with error checking
+    CHIP_ERROR GetMeasurementAccuracy(MeasurementAccuracyStruct & outValue) const;
+    CHIP_ERROR GetCumulativeEnergyImported(Optional<EnergyMeasurementStruct> & outValue) const;
+    CHIP_ERROR GetCumulativeEnergyExported(Optional<EnergyMeasurementStruct> & outValue) const;
+    CHIP_ERROR GetPeriodicEnergyImported(Optional<EnergyMeasurementStruct> & outValue) const;
+    CHIP_ERROR GetPeriodicEnergyExported(Optional<EnergyMeasurementStruct> & outValue) const;
+    CHIP_ERROR GetCumulativeEnergyReset(Optional<CumulativeEnergyResetStruct> & outValue) const;
+
+    // Setters - update values and notify data model
+    CHIP_ERROR SetMeasurementAccuracy(const MeasurementAccuracyStruct & value);
+    CHIP_ERROR SetCumulativeEnergyImported(const Optional<EnergyMeasurementStruct> & value);
+    CHIP_ERROR SetCumulativeEnergyExported(const Optional<EnergyMeasurementStruct> & value);
+    CHIP_ERROR SetPeriodicEnergyImported(const Optional<EnergyMeasurementStruct> & value);
+    CHIP_ERROR SetPeriodicEnergyExported(const Optional<EnergyMeasurementStruct> & value);
+    CHIP_ERROR SetCumulativeEnergyReset(const Optional<CumulativeEnergyResetStruct> & value);
 
-    CHIP_ERROR Init();
-    void Shutdown();
+    DataModel::ActionReturnStatus ReadAttribute(const DataModel::ReadAttributeRequest & request,
+                                                AttributeValueEncoder & encoder) override;
+    CHIP_ERROR Attributes(const ConcreteClusterPath & path, ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder) override;
 
-    CHIP_ERROR Read(const app::ConcreteReadAttributePath & aPath, app::AttributeValueEncoder & aEncoder) override;
+    std::optional<EventNumber> CumulativeEnergySnapshot(const Optional<EnergyMeasurementStruct> & energyImported,
```

Comment:
Does anyone need the event number? I would consider just dropping it to the floor.